tests/misc_env/exec_shell.py

```diff
@@ -0,0 +1,86 @@
+# Module to assist in executing a sequence of command.
+import re
+from logging import getLogger
+from typing import Any
+
+from ceph.ceph_admin import CephAdmin
+from ceph.utils import get_node_by_id
+
+LOG = getLogger(__file__)
+
+
+def lookup_node_id(cluster, pattern: str) -> str:
+    """
+    Return the string with node ID replaced with shortname.
+
+    In this method, the pattern {node:node1} would be replaced with the value of
+    node.shortname.
+
+    Args:
+        cluster:    Ceph cluster instance
+        pattern:    String to be searched for node ID pattern
+
+    Return:
+        String whose node ID's are replaced with shortnames
+    """
+    replaced_string = pattern
+    node_list = re.findall("{node:(.+?)}", pattern)
+
+    for node in node_list:
+        node_name = get_node_by_id(cluster, node).shortname
+        replacement_pattern = "{node:" + node + "}"
+        replaced_string = re.sub(replacement_pattern, node_name, replaced_string)
+
+    return replaced_string
+
+
+def run(ceph_cluster, **kwargs: Any) -> int:
```

Comment:
there is already a function similar to [exec.py](https://github.com/red-hat-storage/cephci/blob/master/tests/misc_env/exec.py) . can we modify this function to add a config to take cephadm as a option to this module for executing  executing ceph_adm shell commands instead of new module. ?